sda/cmd/ingest/ingest.go

```diff
@@ -197,6 +183,23 @@ func main() {
 	<-forever
 }
 
+func (app *Ingest) registerC4GHKey() error {
+	h, err := app.DB.ListKeyHashes()
+	if err != nil {
+		return err
+	}
+	if len(h) == 0 {
```

Comment:
What if `len(h) != 0` but none of the pukeyhashes in the db corresponds to a key in the `ArchiveKeyList`?

This could be a scenario where we add a key to ingest but forget to register its hash to the db and the db keyhash list is not empty.
Wouldn't it then be possible  to ingesting files with an unknown hashkey?

Reply:
First off, that is outside the scope of this issue. And secondly there is no difference with regard to that case compared to the current handling.

But, if we want to safeguard against that kind of misconfiguration we first need to implement liveness checks for the non API components. Since we would need some way to ensure that the deployment will fail visibly instead of just having a pod restart continuously like we have now.